src/lib/utils/Args.ts

```diff
@@ -0,0 +1,167 @@
+import type { Message } from 'discord.js';
+import type * as Lexure from 'lexure';
+import { UserError } from '../errors/UserError';
+import type { ArgumentContext } from '../structures/Argument';
+import type { Command } from '../structures/Command';
+import { err, isOk, Ok, Result } from './Result';
+
+/**
+ * The argument parser to be used in [[Command]].
+ */
+export class Args {
+	/**
+	 * The original message that triggered the command.
+	 */
+	public message: Message;
+
+	/**
+	 * The command that is being running.
+	 */
+	public command: Command;
+	private parser: Lexure.Args;
+	private states: Lexure.ArgsState[] = [];
+
+	public constructor(message: Message, command: Command, parser: Lexure.Args) {
+		this.message = message;
+		this.command = command;
+		this.parser = parser;
+	}
+
+	/**
+	 * Sets the parser to the first token.
+	 */
+	public start(): Args {
+		this.parser.state = {
+			usedIndices: new Set(),
+			position: 0,
+			positionFromEnd: this.parser.parserOutput.ordered.length - 1
+		};
+		return this;
+	}
+
+	/**
+	 * Retrieves the next parameter and parses it. Advances index on success.
+	 * @param type The type of the argument.
+	 * @example
+	 * ```typescript
+	 * // !add 1 2
+	 * const a = await args.pickResult('integer');
+	 * if (!a.success) throw new UserError('AddArgumentError', 'You must write two numbers, but the first one did not match.');
+	 *
+	 * const b = await args.pickResult('integer');
+	 * if (!b.success) throw new UserError('AddArgumentError', 'You must write two numbers, but the second one did not match.');
+	 *
+	 * await message.channel.send(`The result is: ${a.value + b.value}!`);
+	 * // Sends "The result is: 3"
+	 * ```
+	 */
+	public async pickResult<K extends keyof ArgType>(type: K, options: ArgOptions = {}): Promise<Result<ArgType[K], UserError>> {
+		const argument = this.message.client.arguments.get(type);
+		if (!argument) throw new TypeError(`The Argument ${type} was not found.`);
```

Comment:
Probably more accurate as a Range/User Error

also:
```suggestion
		if (!argument) throw new TypeError(`The argument "${type}" was not found.`);
```